test/models/import_test.rb

```diff
@@ -39,15 +40,31 @@ class ImportTest < ActiveSupport::TestCase
   end
 
   test "publishes a valid import" do
-    assert_difference "Transaction.count", 2 do
-      @loaded_import.publish
+    assert_difference("Transaction::Category.count", 2) do
+      assert_difference "Transaction.count", 2 do
+        @loaded_import.publish
+      end
     end
 
     @loaded_import.reload
 
     assert @loaded_import.complete?
   end
 
+  test "publishes a valid import with missing data" do
+    assert_difference("Transaction::Category.count", 1) do
+      assert_difference "Transaction.count", 2 do
+        @loaded_import_with_missing_data.publish
+      end
+    end
```

Comment:
Tiny improvement, believe you can [combine these](https://edgeapi.rubyonrails.org/classes/ActiveSupport/Testing/Assertions.html#method-i-assert_difference) into a single assertion: 

```rb
assert_difference ->{ Transaction::Category.count } => 1, ->{ Transaction.count } => 2 do
  @loaded_import_with_missing_data.publish
end
```

Reply:
Cool, didn't know that, fixed!